### PR TITLE
Feature/92 nav scenario

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-092
* **Review:** By commit
* **Merge strategy:** Merge (no squash)  

## Description
Modifications stemming from a navigation scenario linking modules on develop. 
Contains validity check modifications, regularization of point cloud representation, improvements to read a set of images from file, and additional checks of message content validity. 

## Verification
Unit test suite still passes and was adjusted to the changes

## Documentation
Documentation still holds

## Future work
None

### CODE REVIEW

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp [hunk 1]
```diff
@@ -83,12 +82,11 @@ void PointCloudTriangulation::UpdateState(uint64_t currentSimNanos)
             this->measuredPointCloud.emplace_back(featureLocation);
         }
         this->pointCloudSize = this->numberKeyPoints;
+        this->numberTimesCalled += 1;
```
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp:85]: :large_orange_diamond: Should we update the member variable's comment in the header to say e.g. "Number of times called with valid input data..."

File: src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/cameraTriangulation.h [hunk 2]
```diff
@@ -54,12 +54,12 @@ class CameraTriangulation: public SysModel {
...
     void readMessages();
     void writeMessages(uint64_t currentSimNanos);
-    Eigen::Vector3d triangulation(std::vector<Eigen::Vector3d> knownLocations,
+    Eigen::Vector3d triangulation(Eigen::MatrixXd knownLocations,
                                   std::vector<Eigen::Vector2d> imagePoints,
                                   const Eigen::Matrix3d& cameraCalibrationInverse,
                                   std::vector<Eigen::Matrix3d> dcmCamera) const;
 
-    std::vector<Eigen::Vector3d> pointCloud{}; //!< [-] point cloud
+    Eigen::MatrixXd pointCloud; //!< [-] point cloud
```
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/cameraTriangulation.h]: :large_orange_diamond: We lost the default initializer syntax with this edit '{}'. Was that desired?

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.h [hunk 3]
```diff
@@ -79,7 +79,7 @@ class PointCloudTriangulation: public SysModel {
     Eigen::Matrix3d dcm_C2C1; //!< [-] direction cosine matrix (DCM) from camera frame C1 to camera frame C2
     bool valid; //!< [-] validity flag for point cloud triangulation
     int pointCloudSize{}; //!< [-] number of points in point cloud
-    std::vector<Eigen::Vector3d> measuredPointCloud{}; //!< [-] measured point cloud
+    Eigen::MatrixXd measuredPointCloud; //!< [-] measured point cloud
```
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.h]: :large_orange_diamond: Same here on the default initializer.

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp (hunk 1 above)
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp:85]: Good idea!

File: src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/cameraTriangulation.h (hunk 2 above)
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/cameraTriangulation.h]: I fixed a bunch of these

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.h (hunk 3 above)
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.h]: Same as above

**Pull request merged**